File: 프로그래머스/lv1/12934. 정수 제곱근 판별/정수 제곱근 판별.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

long long solution(long long n) {
    long long x = 0;
    long long i = 1;
    while(1){
        if(n == i * i){
            return (i + 1)*(i + 1);
        }
        else if(n < i * i){
            return -1;
        }
        i++;
    }
 
}

File: 프로그래머스/lv1/12934. 정수 제곱근 판별/정수 제곱근 판별.cs
public class Solution {
    public long solution(long n) {
        long answer = 0;
        for(long i=1; i<= n; i++){
            if(i * i == n){
                answer = (i + 1) * (i + 1);
                break;
            }
            else if(i * i >= n)
            {
                break;
            }
        }
        if(answer == 0){
            return -1;
        }
        else{
            return answer;
        }
    }
}